Src/Newtonsoft.Json/Utilities/TypeExtensions.cs

```diff
@@ -500,7 +500,8 @@ private static bool TestAccessibility(MethodBase member, BindingFlags bindingFla
 
         public static Type[] GetGenericArguments(this Type type)
         {
-            return type.GetTypeInfo().GenericTypeArguments;
```

Comment:
This returns an empty array for generic type definitions which isn't consistent with `Type.GetGenericArguments` thus the need for this fix.